### PR TITLE
Prepare for OneToOne relation to SimHit in MCRecoTrackerHitAssociation

### PR DESCRIPTION
**Pull request by veprbl**
This is needed to support https://github.com/eic/EDM4eic/pull/77

### CODE REVIEW

**Comment by veprbl** [comment]
Updated to include #1392 
Builds with old/new EDM4hep

File: src/algorithms/digi/PhotoMultiplierHitDigi.cc [hunk 1]
```diff
@@ -168,11 +169,16 @@ void PhotoMultiplierHitDigi::process(
 
                 // build `MCRecoTrackerHitAssociation` (for non-noise hits only)
                 if(!data.sim_hit_indices.empty()) {
-                  auto hit_assoc = hit_assocs->create();
-                  hit_assoc.setWeight(1.0); // not used
-                  hit_assoc.setRawHit(raw_hit);
-                  for(auto i : data.sim_hit_indices)
+                  for(auto i : data.sim_hit_indices) {
+                    auto hit_assoc = hit_assocs->create();
+                    hit_assoc.setWeight(1.0); // not used
```
Comment (wdconinc) [inline src/algorithms/digi/PhotoMultiplierHitDigi.cc:174]: Should we keep unitarity and use

```suggestion
                    hit_assoc.setWeight(1.0 / data.size()); // not used
```

File: src/algorithms/pid/IrtCherenkovParticleID.cc [hunk 2]
```diff
@@ -218,17 +219,22 @@ std::map<std::string, std::unique_ptr<edm4eic::CherenkovParticleIDCollection>> e
           for(const auto& hit_assoc : *in_hit_assocs) {
             if(hit_assoc.getRawHit().isAvailable()) {
               if(hit_assoc.getRawHit().id() == raw_hit.id()) {
+#if EDM4EIC_VERSION_MAJOR >= 6
+                  mc_photon = hit_assoc.getSimHit().getMCParticle();
```
Comment (wdconinc) [inline src/algorithms/pid/IrtCherenkovParticleID.cc:223]:
```suggestion
                mc_photon = hit_assoc.getSimHit().getMCParticle();
```

**Review by wdconinc — DISMISSED** [review]

File: src/algorithms/digi/PhotoMultiplierHitDigi.cc [hunk 3]
```diff
@@ -168,11 +169,16 @@ void PhotoMultiplierHitDigi::process(
 
                 // build `MCRecoTrackerHitAssociation` (for non-noise hits only)
                 if(!data.sim_hit_indices.empty()) {
-                  auto hit_assoc = hit_assocs->create();
-                  hit_assoc.setWeight(1.0); // not used
-                  hit_assoc.setRawHit(raw_hit);
-                  for(auto i : data.sim_hit_indices)
+                  for(auto i : data.sim_hit_indices) {
+                    auto hit_assoc = hit_assocs->create();
+                    hit_assoc.setWeight(1.0 / data.size()); // not used
```
Comment (veprbl) [inline src/algorithms/digi/PhotoMultiplierHitDigi.cc:174]:
```suggestion
                    hit_assoc.setWeight(1.0 / data.sim_hit_indices.size()); // not used
```

**Review by wdconinc — APPROVED** [review]

**Pull request merged**